pkg/security/authenticators/basic_authentication.go

```diff
@@ -19,19 +19,25 @@ package authenticators
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/Peripli/service-manager/pkg/types"
-
+	"github.com/Peripli/service-manager/api/osb"
+	"github.com/Peripli/service-manager/pkg/log"
 	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"golang.org/x/crypto/bcrypt"
 
 	httpsec "github.com/Peripli/service-manager/pkg/security/http"
 
 	"github.com/Peripli/service-manager/pkg/web"
 	"github.com/Peripli/service-manager/storage"
 )
 
+//BasicAuthenticatorFunc defines a function which attempts to authenticate a basic auth request
+type BasicAuthenticatorFunc func(request *web.Request, repository storage.Repository, username, password string) (*web.UserContext, httpsec.Decision, error)
+
 // Basic for basic security
 type Basic struct {
-	Repository storage.Repository
+	Repository             storage.Repository
+	BasicAuthenticatorFunc BasicAuthenticatorFunc
```

Comment:
This struct was from before. We need it to inject the storage in some way - we cannot provide into the Authenticate function as it will not implement Authenticator then.